src/commands/aem/rde/install.js

```diff
@@ -192,13 +195,33 @@ class DeployCommand extends BaseCommand {
           });
         } else {
           type = guessedTypes[0];
+          this.doLog(chalk.yellow(`No --type provided, using ${type}`));
+        }
+      }
+
+      // when no path was defined explicitly, we also try to guess it
+      if ((type === 'content-file' || type === 'content-xml') && !flags.path) {
+        if (isLocalFile && inputPath.includes('/' + JCR_ROOT + '/')) {
+          const guessedPath = guessPath(inputPath);
+          if (guessedPath) {
+            flags.path = guessedPath;
+            this.doLog(
+              chalk.yellow(
+                `No --path provided, repository path was set to ${flags.path}`
+              )
+            );
+          }
         }
       }
     } catch (err) {
       this.doLog(err);
       return;
     }
 
+    if (!flags.path) {
+      throw new validationCodes.MISSING_CONTENT_PATH();
+    }
```

Comment:
Does  this not error out for any content type other than "content-xml" and "content-file"?

Reply:
woops true, moved it to else case. good that I start working on test coverage on that class... sorry.